packages/voyager-keycloak/src/KeycloakSecurityService.ts

```diff
@@ -90,9 +96,29 @@ export class KeycloakSecurityService implements SecurityService {
     }
   }
 
-  public async validateToken(token: string): Promise<boolean> {
-    const tokenObject = getTokenObject(token)
-    const result = await this.keycloak.grantManager.validateToken(tokenObject, 'Bearer')
-    return (result === tokenObject) ? true : false
+  public async onSubscriptionConnect(connectionParams: any, webSocket: any, context: any): Promise<any> {
+    const header = connectionParams.Authorization
+                  || connectionParams.authorization
```

Comment:
backward compatibility? I think we should send token here and detect if object is string

Reply:
I was going to say if we interface connectionParams then we can type it properly and future proof it for whatever else we might need to add.

Reply:
The way we parse this token is compatible with GraphQL playground and other similar technologies where there is usually a box to pass in some extra http headers. The same values in there are also passed in the subscription payload.

With regards to connectionParams, I'm not sure about trying to interface that as it literally can be anything. The client can pass objects, raw values like Strings and numbers, they can pass nothing at all. We can define an interface for what the connectionParams should look like when using auth, and then cast to it. That will allow us to have a nice development experience but you still need to do some checks at runtime.

Reply:
Sorry for that. I totally forgot that we wanted to really keep GraphQL playground compatibility.